group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorShard.java

```diff
@@ -503,6 +547,7 @@ public void onLoaded(MetadataImage newImage) {
         MetadataDelta emptyDelta = new MetadataDelta(newImage);
         groupMetadataManager.onNewMetadataImage(newImage, emptyDelta);
         offsetMetadataManager.onNewMetadataImage(newImage, emptyDelta);
+        coordinatorMetrics.activateMetricsShard(metricsShard);
```

Comment:
It's probably not a concern for this PR, but it would be nice if the activate and deactivate could be in the same file. 
I think any coordinator shard could have an unload method for example.